pkg/infrastructure/azure/dns.go

```diff
@@ -85,25 +84,13 @@ func createDNSEntries(ctx context.Context, in clusterapi.InfraReadyInput, extLBF
 		return fmt.Errorf("failed to create session: %w", err)
 	}
 	subscriptionID := session.Credentials.SubscriptionID
-	cloudConfiguration := session.CloudConfig
 
-	recordSetClient, err := armdns.NewRecordSetsClient(subscriptionID, session.TokenCreds,
-		&arm.ClientOptions{
-			ClientOptions: policy.ClientOptions{
-				Cloud: cloudConfiguration,
-			},
-		},
-	)
+	opts.APIVersion = "2018-05-01" // Azure-Stack-compatible API version
```

Comment:
Should we check if opts != nil and/or if this is an ASH install?

Reply:
I think it was addressed in the next commit: https://github.com/openshift/installer/pull/9645/commits/9ace677ee124b11898903615c14a7e2c27dace40, right?

Reply:
Would probably have been better to move this line into the next commit.